variables.tf

```diff
@@ -148,6 +148,22 @@ variable "labels" {
   description = "Labels to be attached to the buckets"
   type        = map(string)
   default     = {}
+  # Validations based on https://cloud.google.com/storage/docs/tags-and-labels#bucket-labels
+  validation {
+    condition = alltrue([
```

Comment:
Since upstream APIs can change this would be hard to keep upto date in the module. The provider seems to have some prior art doing this, my suggestion would be to upstream this check there
https://github.com/hashicorp/terraform-provider-google/blob/678449a1b0559d19b430ed760056734ea25c5e96/google/services/cloudfunctions/resource_cloudfunctions_function.go#L48